src/state/transactions/hooks.tsx

```diff
@@ -72,16 +72,31 @@ export function useAllTransactions(): { [txHash: string]: TransactionDetails } {
   return transactionsFilteredByConnectedAccount
 }
 
+/**
+ * Returns whether a transaction happened in the last day (86400 seconds * 1000 milliseconds / second)
+ * @param tx to check for recency
+ */
+export function isTransactionRecent(tx: TransactionDetails): boolean {
+  return new Date().getTime() - tx.addedTime < 86_400_000
+}
+
 // returns whether a token has a pending approval transaction
 export function useHasPendingApproval(tokenAddress?: string): boolean {
   const allTransactions = useAllTransactions()
-  return typeof tokenAddress !== 'string'
-    ? false
-    : Object.keys(allTransactions).some((hash) => {
-        if (allTransactions[hash]?.receipt) {
+  return useMemo(() => {
+    return (
+      typeof tokenAddress === 'string' &&
+      Object.keys(allTransactions).some((hash) => {
+        const tx = allTransactions[hash]
+        if (!tx) return false
+        if (tx.receipt) {
           return false
         } else {
-          return allTransactions[hash]?.approvalOfToken === tokenAddress
+          const { approvalOfToken } = tx
+          if (!approvalOfToken) return false
+          return approvalOfToken === tokenAddress && isTransactionRecent(tx)
```

Comment:
Can you explain this? You are adding another constraint , so your funciton is returning less likely a true.

But we have the problem that it is not returning true, although it should return true. Hence, I think that your fix will not improve the situation.

Reply:
We should make that a user has a pending approval less likely to be true. Maybe I'm missing something, but I will re-check everything.

Reply:
Ah I see.

Reply:
Also the spender could be the problem: 
https://github.com/Uniswap/uniswap-interface/blob/8fd894f2d1b03921c5519c050aed164343c47fb1/src/state/transactions/hooks.tsx#L80

Maybe we should use uniswaps new logic for it...